source/windows/windows_pki_utils.c

```diff
@@ -568,16 +568,17 @@ int aws_import_key_pair_to_cert_context(
         return AWS_OP_ERR;
     }
 
+    if (aws_array_list_init_dynamic(&private_keys, alloc, 1, sizeof(struct aws_byte_buf))) {
+        aws_array_list_clean_up(&certificates);
+        return AWS_OP_ERR;
+    }
+
```

Comment:
Our general approach to clean_up() functions is: they should tolerate being passed zeroed-out stuff, so we can naively clean stuff up even if it wasn't fully initialized, to avoid weird fragile cleanup bugs like this...

But aws_cert_chain_clean_up() was hitting an ASSERT because it tries to iterate over an array-list, but the array-list is zeroed out. And when `aws_array_list_size()` is called, it freaks about because the array-list isn't initialized.

I'd propose a different solution: aws_cert_chain_clean_up() should tolerate a zeroed-out list, and the easiest way to achieve that is to make aws_array_list_size() tolerate being called on a zeroed out list. 

Instead of: ~AWS_PRECONDITION(aws_array_list_is_valid(list));~
We can do: `AWS_PRECONDITION(AWS_IS_ZEROED(*list) || aws_array_list_is_valid(list));`

Reply:
this goes against our general rule of: the only functions that should tolerate being passed NULL, or zeroed-out-but-uninitialized struct are `clean_up() / release() / destroy()` functions

But since cleaning up an `aws_array_list` requires the user to first iterate over its members and clean them up, I think we should make an exception and let `aws_array_list_size()`

Reply:
Thanks, updated.